cfgov/unprocessed/js/owning-a-home/modules/explore-rates.js

```diff
@@ -0,0 +1,1224 @@
+ */
+function resultWarning() {
+  chart.stopLoading( 'error' );
+  $( '#chart-section' ).addClass( 'warning' ).append( template.resultAlert );
+}
+
+function resultFailWarning() {
+  chart.stopLoading( 'error' );
+  $( '#chart-section' ).addClass( 'warning' ).append( template.failAlert );
+}
+
+function downPaymentWarning() {
+  $( '#loan-amt-inputs' ).append( template.dpWarning );
+}
+
+
+/**
+ * Remove alerts and warnings
+ */
+function removeAlerts() {
+  if ( $( '.result-alert' ) ) {
+    $( '#chart' ).removeClass( 'warning' );
+    $( '.result-alert' ).not( '.credit-alert' ).remove();
+    $( '#dp-alert' ).remove();
+  }
+}
+
+
+function removeCreditScoreAlert() {
+  if ( $( '.credit-alert' ) || $( '.rangeslider__handle' ).hasClass( 'warning' ) ) {
+    $( '.rangeslider__handle' ).removeClass( 'warning' );
+    $( '.credit-alert' ).remove();
+  }
+}
+
+/**
+ * Add commas to numbers where appropriate.
+ * @param {string} value - Old value where commas will be added.
+ * @returns {string} value - Value with commas and no dollar sign.
+ */
+function addCommas( value ) {
+  value = unFormatUSD( value );
+  value = formatUSD( value, { decimalPlaces: 0 } )
+    .replace( '$', '' );
+  return value;
+}
+
+/**
+ * Initialize the range slider. http://andreruffert.github.io/rangeslider.js/
+ * @param {Function} cb - Optional callback.
+ * @returns {*} - Value of callback invocation.
+ */
+function renderSlider( cb ) {
+
+  $( '#credit-score' ).rangeslider( {
+    polyfill:    false,
+    rangeClass:  'rangeslider',
+    fillClass:   'rangeslider__fill',
+    handleClass: 'rangeslider__handle',
+    onInit:      function() {
+      slider.update();
+    },
+    onSlide:     function( position, value ) {
+      slider.update();
+    },
+    onSlideEnd:  function( position, value ) {
+      params.update();
+      if ( params['credit-score'] < 620 ) {
+        removeAlerts();
+        scoreWarning();
+      } else {
+        updateView();
+        removeCreditScoreAlert();
+      }
+    }
+  } );
+
+  if ( cb ) {
+    return cb();
+  }
+
+  return UNDEFINED;
+
+}
+
+/**
+ * Render chart data in an accessible format.
+ * @param {Object} data - Data processed from the API.
+ */
+function renderAccessibleData( data ) {
+  var $tableHead = $( '#accessible-data .table-head' );
+  var $tableBody = $( '#accessible-data .table-body' );
+
+  $tableHead.empty();
+  $tableBody.empty();
+
+  $.each( data.labels, function( index, value ) {
+    $tableHead.append( '<th>' + value + '</th>' );
+  } );
+
+  $.each( data.vals, function( index, value ) {
+    $tableBody.append( '<td>' + value + '</td>' );
+  } );
+}
+
+/**
+ * Render (or update) the Highcharts chart.
+ * @param  {Object} data Data processed from the API.
+ * @param  {Function} cb Optional callback.
+ * @returns {*} Value of callback invocation or undefined.
+ */
+function renderChart( data, cb ) {
+
+  if ( chart.isInitialized ) {
+
+    Highcharts.setOptions( {
```

Comment:
The chart doesn't seem to be getting updated for me when new data is received:

<img width="737" alt="screen shot 2017-08-10 at 7 47 55 am" src="https://user-images.githubusercontent.com/778171/29176304-4e716312-7da0-11e7-9ad9-89cb140cc386.png">


It does update if I call `update` on the chart itself instead of `Highcharts.setOptions`, though.

<img width="717" alt="screen shot 2017-08-10 at 7 45 30 am" src="https://user-images.githubusercontent.com/778171/29176313-56d0fd9c-7da0-11e7-82b5-17711c9ecbf5.png">